Source/LinqToDB/LinqExtensions.Delete.cs

```diff
@@ -74,6 +72,32 @@ public static IEnumerable<TSource> DeleteWithOutput<TSource>(this IQueryable<TSo
 						null,
 						MethodHelper.GetMethodInfo(DeleteWithOutput, source),
 						currentSource.Expression))
+				.AsAsyncEnumerable();
+		}
+
+		/// <summary>
+		/// Deletes records from source query into target table asynchronously and returns deleted records.
+		/// </summary>
+		/// <typeparam name="TSource">Source query record type.</typeparam>
+		/// <param name="source">Source query, that returns data for delete operation.</param>
+		/// <param name="token">Optional asynchronous operation cancellation token.</param>
+		/// <returns>Array of records.</returns>
+		/// <remarks>
+		/// Database support:
+		/// <list type="bullet">
+		/// <item>SQL Server 2005+</item>
+		/// <item>Firebird 2.5+ (prior version 5 returns only one record; database limitation)</item>
+		/// <item>PostgreSQL</item>
+		/// <item>SQLite 3.35+</item>
+		/// <item>MariaDB 10.0+ (doesn't support multi-table statements; database limitation)</item>
+		/// </list>
+		/// </remarks>
+		[Obsolete("Will be removed in l2db 7.0")]
```

Comment:
I suggest a longer obsolete message that explains to the developer the replacement syntax when they are fixing their code.

```suggestion
		[Obsolete("Will be removed in linq2db 7.0. Please remove the cancellation token and use await foreach or ToListAsync instead.")]
```

Not sure exactly, but something like that.

Reply:
I would argue it doesn't really matter, since the only way to call this method is: 
```cs
LinqExtensions.DeleteWithOutputAsync(dbContext, ...);
```
since the `this` has been removed.

Reply:
So it’s only there for binary compatibility at all then?

Reply:
Correct